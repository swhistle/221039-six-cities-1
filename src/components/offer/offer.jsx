import React from "react";
import PropTypes from "prop-types";

export class OfferComponent extends React.PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const {offer} = this.props;

    if (offer) {
      return <main className="page__main page__main--property">
        <section className="property">
          <div className="property__gallery-container container">
            <div className="property__gallery">
              {
                offer.images.map((item, index) =>
                  <div key={`image-${index}`} className="property__image-wrapper">
                    <img className="property__image" src={item} alt="Photo studio"/>
                  </div>)
              }
            </div>
          </div>
          <div className="property__container container">
            <div className="property__wrapper">
              {
                offer.is_premium ?
                  <div className="property__mark">
                    <span>Premium</span>
                  </div> : ``
              }
              <div className="property__name-wrapper">
                <h1 className="property__name">
                  {offer.title}
                </h1>
                <button className="property__bookmark-button button" type="button">
                  <svg className="property__bookmark-icon" width="31" height="33">
                    <use xlinkHref="#icon-bookmark"></use>
                  </svg>
                  <span className="visually-hidden">To bookmarks</span>
                </button>
              </div>
              <div className="property__rating rating">
                <div className="property__stars rating__stars">
                  <span style={{width: `96%`}}></span>
                  <span className="visually-hidden">Rating</span>
                </div>
                <span className="property__rating-value rating__value">{offer.rating}</span>
              </div>
              <ul className="property__features">
                <li className="property__feature property__feature--entire">
                  {offer.type}
                </li>
                <li className="property__feature property__feature--bedrooms">
                  {offer.bedrooms} Bedrooms
                </li>
                <li className="property__feature property__feature--adults">
                  Max {offer.max_adults} adults
                </li>
              </ul>
              <div className="property__price">
                <b className="property__price-value">&euro;{offer.price}</b>
                <span className="property__price-text">&nbsp;night</span>
              </div>
              <div className="property__inside">
                <h2 className="property__inside-title">What&apos;s inside</h2>
                <ul className="property__inside-list">
                  {
                    offer.goods.map((good, index) =>
                      <li key={`property-${index}`} className="property__inside-item">
                        {good}
                      </li>)
                  }
                </ul>
              </div>
              <div className="property__host">
                <h2 className="property__host-title">Meet the host</h2>
                <div className="property__host-user user">
                  <div className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper">
                    <img className="property__avatar user__avatar" src={offer.host.avatar_url} width="74" height="74" alt="Host avatar"/>
                  </div>
                  <span className="property__user-name">
                    {offer.host.name}
                  </span>
                  {
                    offer.host.is_pro ?
                      <span className="property__user-status">
                        Pro
                      </span> : ``
                  }
                </div>
                <div className="property__description">
                  <p className="property__text">
                    {offer.description}
                  </p>
                </div>
              </div>
            </div>
          </div>
        </section>
      </main>;
    }

    return null;
  }
}

OfferComponent.propTypes = {
  offer: PropTypes.shape({
    id: PropTypes.number,
    type: PropTypes.string,
    title: PropTypes.string,
    price: PropTypes.number,
    rating: PropTypes.number,
    images: PropTypes.arrayOf(PropTypes.string),
    bedrooms: PropTypes.number,
    goods: PropTypes.arrayOf(PropTypes.string),
    host: PropTypes.shape({
      name: PropTypes.string
    }),
    description: PropTypes.string
  })
};
