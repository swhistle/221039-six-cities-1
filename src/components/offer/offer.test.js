import React from 'react';
import renderer from 'react-test-renderer';

import {OfferComponent} from "./offer.jsx";

const TEST_OFFER = {
  id: 1,
  type: `Private room`,
  title: `Wood and stone place`,
  images: [``],
  price: 80,
  rating: 4,
  bedrooms: 1,
  goods: [],
  host: {
    name: `host`,
  },
  description: `description`
};

it(`OfferComponent correct renders`, () => {
  const offer = renderer
    .create(
        <OfferComponent offer={TEST_OFFER}/>
    ).toJSON();

  expect(offer).toMatchSnapshot();
});
