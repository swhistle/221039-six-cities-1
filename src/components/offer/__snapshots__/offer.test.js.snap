// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferComponent correct renders 1`] = `
<main
  className="page__main page__main--property"
>
  <section
    className="property"
  >
    <div
      className="property__gallery-container container"
    >
      <div
        className="property__gallery"
      >
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src=""
          />
        </div>
      </div>
    </div>
    <div
      className="property__container container"
    >
      <div
        className="property__wrapper"
      >
        
        <div
          className="property__name-wrapper"
        >
          <h1
            className="property__name"
          >
            Wood and stone place
          </h1>
          <button
            className="property__bookmark-button button"
            type="button"
          >
            <svg
              className="property__bookmark-icon"
              height="33"
              width="31"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="property__rating rating"
        >
          <div
            className="property__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "96%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
          <span
            className="property__rating-value rating__value"
          >
            4
          </span>
        </div>
        <ul
          className="property__features"
        >
          <li
            className="property__feature property__feature--entire"
          >
            Private room
          </li>
          <li
            className="property__feature property__feature--bedrooms"
          >
            1
             Bedrooms
          </li>
          <li
            className="property__feature property__feature--adults"
          >
            Max 
             adults
          </li>
        </ul>
        <div
          className="property__price"
        >
          <b
            className="property__price-value"
          >
            €
            80
          </b>
          <span
            className="property__price-text"
          >
             night
          </span>
        </div>
        <div
          className="property__inside"
        >
          <h2
            className="property__inside-title"
          >
            What's inside
          </h2>
          <ul
            className="property__inside-list"
          />
        </div>
        <div
          className="property__host"
        >
          <h2
            className="property__host-title"
          >
            Meet the host
          </h2>
          <div
            className="property__host-user user"
          >
            <div
              className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
            >
              <img
                alt="Host avatar"
                className="property__avatar user__avatar"
                height="74"
                width="74"
              />
            </div>
            <span
              className="property__user-name"
            >
              host
            </span>
            
          </div>
          <div
            className="property__description"
          >
            <p
              className="property__text"
            >
              description
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>
</main>
`;
